src/ploomber/spec/dagspec.py

```diff
@@ -467,8 +501,8 @@ def _to_dag(self):
             if attr in self:
                 setattr(
                     dag, attr,
-                    dotted_path.DottedPath(self[attr],
-                                           lazy_load=self._lazy_import))
+                    dotted_path.DottedPath(
+                        self[attr], lazy_load=self._import_mode != 'eager'))
```

Comment:
I am not sure `DottedPath` needs to know more than a boolean: "should I load this module now or not?" so I've preserved the lazy load logic where we don't need to explicitly skip the loading entirely.